### ISSUE TITLE
Issues when modifying an employee's password

### ISSUE THREAD

**Issue by Julievrz** [PrestaShop/PrestaShop#22429]
#### Describe the bug

When modifying an employee's password, multiple issues happened.

**1.** When entering a very weak password such as "123456789", a message appears under the field, saying that the password entered is "Fabulous". Regarding this first issue, it might be linked to this issue (https://github.com/PrestaShop/PrestaShop/issues/11752). @PierreRambaud is your issue related to employees' passwords too?

**2.** When entering a password containing invalid characters or a password that is less than 8 characters, nothing happens when we click on "Save". 

**3.** When the content of the fields "New password" and "Confirm password", nothing happens when we click on "Save".

#### Expected behavior

**1.** A simple password like 123456789 should not be considered as "Fabulous".
**2.** An error message should appear to let the user know that the password entered contains invalid characters or is too short.
**3.** An error message should appear to let the user know that the content in the "New password" and "Confirm password" fields are not the same.

#### Steps to Reproduce

1. Go to Advanced Parameters > Team 
2. Click on Edit.
3. Scroll down to "Change password..."
4. Test the previous behaviors and see the error

#### Additional information

* PrestaShop version: 1.7.7
* PHP version: 7.3.24

**Comment by Julievrz** [comment]
@hibatallahAouadni

**Comment by hibatallahAouadni** [comment]
Hello @Julievrz 

> Regarding this first issue, it might be linked to this issue (#11752). @PierreRambaud is your issue related to employees' passwords too?

I checked @PierreRambaud 's PR, and I believe it fixes the issue with Employee passwords too #12824 
But let's wait for @PierreRambaud confirmation :sweat_smile:

**Comment by Julievrz** [comment]
@hibatallahAouadni Great, thanks! :) What about the other issues? @khouloudbelguith was able to reproduce them.

**Comment by Julievrz** [comment]
https://github.com/PrestaShop/PrestaShop/pull/22045#issuecomment-741856612

**Comment by PierreRambaud** [comment]
Yes we want to add an admin section to improve the password policy for the front office.
But didn't have the time to finish it :( 
I'm closing it since it's a duplicate

**Comment by Julievrz** [comment]
HI @PierreRambaud 

But what about the issue 2) and 3)? Are they duplicate too? 🤔 

2. When entering a password containing invalid characters or a password that is less than 8 characters, nothing happens when we click on "Save".

3. When the content of the fields "New password" and "Confirm password", nothing happens when we click on "Save".

**Comment by marwachelly** [comment]
Hi @Julievrz 

> HI @PierreRambaud
> 
> But what about the issue 2) and 3)? Are they duplicate too? 
> 
> 1. When entering a password containing invalid characters or a password that is less than 8 characters, nothing happens when we click on "Save".
> 2. When the content of the fields "New password" and "Confirm password", nothing happens when we click on "Save".

I reproduced the same issue. @hibatallahAouadni could you please reopen the issue.

**Comment by matks** [comment]
> HI @PierreRambaud
> 
> But what about the issue 2) and 3)? Are they duplicate too? 🤔
> 
> 1. When entering a password containing invalid characters or a password that is less than 8 characters, nothing happens when we click on "Save".
> 2. When the content of the fields "New password" and "Confirm password", nothing happens when we click on "Save".

Items 2) and 3) are not valid too. You can mark them as bugs.

**Comment by hibatallahAouadni** [comment]
Hello everyone,

I reproduce the issue even with PS1767, see the attached screen record below:

https://watch.screencastify.com/v/qfzhQCjfSHlQXpkxQzlh

Thanks!

**Comment by boo-code** [comment]
All three points look resolved on `9.2.x`. Evidence for each, since they were fixed by different changes rather than by one.

**2 and 3, the silent saves.** I submitted `ChangePasswordType` directly and read the form errors back:

```
mismatch       valid=false  errors=["The values do not match."]
3-character    valid=false  errors=["This field cannot be shorter than 8 characters"]
```

So the form now rejects both and carries a message for each, rather than doing nothing. The confirmation field is a `RepeatedType`, which produces "The values do not match." by itself, and the length comes from a `Length` constraint built from `PasswordPolicyConfiguration` rather than a hard-coded number. The second field also carries `data-invalid-password` with "The confirmation password doesn't match.", so the mismatch is reported before the round trip as well.

**1, the strength rating.** The meter is no longer the old scale. `admin-dev/themes/new-theme/js/components/change-password-handler.ts` scores with `@zxcvbn-ts/core` loaded with the common and English dictionaries, and `123456789` is a dictionary entry, so it scores at the bottom rather than as "Fabulous". The minimum acceptable score is configurable and reaches the field as `data-minscore`.

One caveat on what I checked: this is the form and the scoring code, not a click-through in the browser, so if any of you can still see the old behaviour on a current version it is worth saying which of the three and on what version, because the code paths for them are now separate.

Otherwise this looks closable.
